### DISCUSSION TITLE
Literal coercion

### DISCUSSION THREAD

**Discussion by vlulla** [duckdb/duckdb#8151] [Ideas]
Google's BigQuery (BQ) has this convenience feature called [literal coercion](https://cloud.google.com/bigquery/docs/reference/standard-sql/conversion_rules#literal_coercion) which is very handy for date/datetime/timestamp arithmetic which allows you to write statements like `select '2023-02-23' + interval 6 month`! While this simple statement appears trivial I have found this convenience feature quite valuable in exploratory data analysis in BQ. So, I was wondering if it is possible to do something like this in duckdb.

I'm hopeful that this should be possible because when I type, in duckdb, `select '2023-02-23' + interval 6 month` I receive the error `Error: Binder Error: No function matches the given name and argument types '+(VARCHAR, INTERVAL)'. You might need to add explicit type casts.` So, I'm assuming that there's a function dispatch somewhere where this special case of `+(VARCHAR, INTERVAL)` (or possibly `+(VARCHAR, date)`/`+(VARCHAR,datetime)`/`+(VARCHAR,timestamp)` and also for minus) can be introduced which will make duckdb even more awesome for olap!

Anyways, i love using duckdb! Thanks for making such an awesome olap database!

PS: I'll be happy to help with testing/contributing for this...but unfortunately i'm not too familiar with duckdb architecture. I'm happy to learn if you can point me to docs/wiki which explains, even at a very high level, technical details of function dispatch.

**Comment by Tishj** [comment]
Yes that's correct, in DuckDB the function overload gets selected based on the types of the input parameters.
using
```sql
select '2023-02-23'::TIMESTAMP + interval 6 month;
```
This does what you want it to do :)

**Reply by vlulla** [reply]
Indeed! 

As far as I'm aware this literal coercion is only available in BQ. And, this literal coercion works for only one case...where a string can be represented as date/datetime/time/timestamp! As I mentioned in my suggestion...I think this is a convenience feature which will make, at least in my opinion, working with duckdb a _little more_ fun. Anyways, I don't have strong opinions if this is implemented or not! Thanks for your comment @Tishj .
